cds-plugin.js

```diff
@@ -3,29 +3,70 @@
 
 const cds = require("@sap/cds");
 const cdsPackage = require("@sap/cds/package.json");
-const { initializeFeatures } = require("./src/singleton");
+const { initializeFeatures, getFeaturesKeys, getFeatureValue } = require("./src/singleton");
 
-const activate = async () => {
-  const envFeatureToggles = cds.env.featureToggles;
-  if (envFeatureToggles?.config || envFeatureToggles?.configFile) {
-    // TODO this is currently done in package.json, because "cds build" ignores it otherwise. However, it should happen
-    //  dynamically.
-    // cds.env.requires["FeatureService"] = { model: "@cap-js-community/feature-toggle-library" };
-
-    if (Array.isArray(envFeatureToggles.serviceAccessRoles)) {
-      cds.on("loaded", (csn) => {
-        if (csn.definitions.FeatureService) {
-          csn.definitions.FeatureService["@requires"] = envFeatureToggles.serviceAccessRoles;
-        }
-      });
+const FEATURE_KEY_REGEX = /\/fts\/([^\s/]+)$/;
+
+const _overwriteServiceAccessRoles = (envFeatureToggles) => {
+  if (!Array.isArray(envFeatureToggles.serviceAccessRoles)) {
+    return;
+  }
+  cds.on("loaded", (csn) => {
+    if (csn.definitions.FeatureService) {
+      csn.definitions.FeatureService["@requires"] = envFeatureToggles.serviceAccessRoles;
     }
+  });
+};
 
-    // TODO for the "cds build" use case, this initialize makes no sense
-    await initializeFeatures({
-      config: envFeatureToggles.config,
-      configFile: envFeatureToggles.configFile,
-    });
+const _registerFeatureProvider = () => {
+  if (!cds.env.requires?.toggles || !cds.middlewares?.before) {
+    return;
+  }
+  const cdsFeatures = getFeaturesKeys().reduce((result, key) => {
+    const match = FEATURE_KEY_REGEX.exec(key);
+    if (match) {
+      const feature = match[1];
+      result.push([key, feature]);
+    }
+    return result;
+  }, []);
+  if (cdsFeatures.length === 0) {
+    return;
+  }
+  const contextAuthIndex = cds.middlewares.before.findIndex((entry) => entry.name === "cds_context_auth");
```

Comment:
With cds 7.5 Daniel would like to eliminate the cds_context_auth middleware.

There's also an easier way to register the middleware with `cds.middlewares.add (cds_feature_provider, {before:'ctx_model'})` instead of the splicing you currently do.

Alternatively, you could also register it as last middleware to be run before the protocol adapter with `cds.middlewares.before.push(cds_feature_provider)` or `cds.middlewares.add(cds_feature_provider)`

Reply:
Being last doesn't work, because the model is different depending in the active features. Max and Oli tried the before: xxx syntax and it didn't work for them. But I can try it too... I think this syntax is also undocumented for now.

Reply:
@johannes-vogel I will use this approach. It works fine for me. I also think that _before_ `ctx_model` is a better  placement than _after_ `ctx_auth`. The feature vector should be about the last thing you take care of before preparing the model.

Reply:
My previous alternative (registering as last middleware) was stupid of course as the toggles must be respected in cds.context.model.

Reply:
I was wrong about the .add syntax being undocumented... I had looked in middleware customization
https://cap.cloud.sap/docs/node.js/middlewares#customization

But it was in the cds-serve part
https://cap.cloud.sap/docs/node.js/cds-serve#cds-middlewares